app/src/main/java/com/firebase/uidemo/database/ChatActivity.java

```diff
@@ -70,8 +75,10 @@ protected void onCreate(Bundle savedInstanceState) {
         mManager.setReverseLayout(false);
 
         mMessages = (RecyclerView) findViewById(R.id.messagesList);
-        mMessages.setHasFixedSize(false);
+        mMessages.setHasFixedSize(true);
```

Comment:
@samtstern since adding architecture components was all about improving performance for my app, I finally decided to figure out what this flag does. Basically, if you can guarantee that the recyclerview's size isn't affected by its children, then it doesn't have to invalidate itself whenever the data changes. This is especially important for a chat app since messages are coming in all the time.

To get this to work, I changed the recyclerview height to `match_parent` instead of `wrap_content`.